docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+| **Possible values** | &gt;= `"0s"`      |
+
+The `timeout_commit` represents the minimum interval between the commit of a
+block until the start of the next height of consensus.
+It gives the node a chance to gather additional precommits for the committed
+block, more than the mandatory +2/3 precommits required to commit a block.
+The more precommits are gathered for a block, the greater are the safety
+guarantees and the easier is to detect misbehaving validators.
+
+The `timeout_commit` is not a required component of the consensus algorithm,
+meaning that there are no liveness implications if it is set to `0s`.
+But it does have implications in the way the application rewards validators.
+
+### consensus.double_sign_check_height
+
+How many blocks to look back to check the existence of the node's consensus votes before joining consensus.
+
+```toml
+double_sign_check_height = 0
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+When non-zero, the validator will panic upon restart if the validator's current
+consensus key was used to sign any precommit message for the last
+`double_sign_check_height` blocks.
+If this happens, the validators should stop the state machine, wait for some
+blocks, and then restart the state machine again.
+
+### consensus.skip_timeout_commit
+
+Start the next height as soon as the node gathers all the mandatory +2/3 precommits for a block.
+
+```toml
+skip_timeout_commit = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+Setting `skip_timeout_commit` to `true` has the similar effect as setting [`timeout_commit`](#consensustimeout_commit) to `"0s"`.
+
+### consensus.create_empty_blocks
+
+Propose empty blocks if the validator's mempool does not have any transaction.
+
+```toml
+create_empty_blocks = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+When set to `true`, empty blocks are produced and proposed to indicate that the
+chain is still operative.
+When set to `false`, blocks are not produced or proposed while there are no
+transactions in the validator's mempool.
+
+Notice that empty blocks are still proposed whenever the application hash
+(`app_hash`) has been updated.
+
+This is more relevant for networks with a low volume number of transactions.
+
+### consensus.create_empty_blocks_interval
+
+How long a validator should wait before proposing an empty block.
+
+```toml
+create_empty_blocks_interval = "0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+If there are no transactions in the validator's mempool, the validator
+waits for `create_empty_blocks_interval` before producing and proposing an
+empty block (with no transactions).
```

Comment:
Does it also wait `create_empty_blocks_interval` if there is a change in `app_hash`?

Reply:
Nope.

Reply:
But this is mentioned in the second paragraph below.

Reply:
Addressed